generic/htx_test.py

```diff
@@ -83,23 +86,50 @@ def setup_htx(self):
             if not smm.check_installed(pkg) and not smm.install(pkg):
                 self.cancel("Can not install %s" % pkg)
 
-        url = "https://github.com/open-power/HTX/archive/master.zip"
-        tarball = self.fetch_asset("htx.zip", locations=[url], expire='7d')
-        archive.extract(tarball, self.teststmpdir)
-        htx_path = os.path.join(self.teststmpdir, "HTX-master")
-        os.chdir(htx_path)
-
-        exercisers = ["hxecapi_afu_dir", "hxedapl", "hxecapi", "hxeocapi"]
-        for exerciser in exercisers:
-            process.run("sed -i 's/%s//g' %s/bin/Makefile" % (exerciser,
-                                                              htx_path))
-
-        build.make(htx_path, extra_args='all')
-        build.make(htx_path, extra_args='tar')
-        process.run('tar --touch -xvzf htx_package.tar.gz')
-        os.chdir('htx_package')
-        if process.system('./installer.sh -f'):
-            self.fail("Installation of htx fails:please refer job.log")
+        if self.run_type == 'git':
+            url = "https://github.com/open-power/HTX/archive/master.zip"
+            tarball = self.fetch_asset("htx.zip", locations=[url], expire='7d')
+            archive.extract(tarball, self.teststmpdir)
+            htx_path = os.path.join(self.teststmpdir, "HTX-master")
+            os.chdir(htx_path)
+
+            exercisers = ["hxecapi_afu_dir", "hxedapl", "hxecapi", "hxeocapi"]
+            for exerciser in exercisers:
+                process.run("sed -i 's/%s//g' %s/bin/Makefile" % (exerciser,
+                                                                  htx_path))
+
+            build.make(htx_path, extra_args='all')
+            build.make(htx_path, extra_args='tar')
+            process.run('tar --touch -xvzf htx_package.tar.gz')
+            os.chdir('htx_package')
+            if process.system('./installer.sh -f'):
+                self.fail("Installation of htx fails:please refer job.log")
+        else:
+            dist_name = detected_distro.name.lower()
+            if dist_name == 'suse':
+                dist_name = 'sles'
+            rpm_check = "htx%s%s" % (dist_name, detected_distro.version)
+            skip_install = False
+            ins_htx = process.system_output(
+                'rpm -qa | grep htx', shell=True, ignore_status=True).decode()
+            if ins_htx:
+                if not smm.check_installed(rpm_check):
+                    self.log.info("Clearing existing HTX rpm")
+                    process.system('rpm -e %s' %
+                                   ins_htx, shell=True, ignore_status=True)
+                    if os.path.exists('/usr/lpp/htx'):
+                        shutil.rmtree('/usr/lpp/htx')
+                else:
+                    self.log.info("Using existing HTX")
+                    skip_install = True
+            if not skip_install:
+                rpm_loc = self.params.get('rpm_link', default=None)
```

Comment:
i did not like this idea as htx link we don't get in public url

Reply:
@PraveenPenguin I think this is fine, since the idea of building an htx rpm itself is public as per 
https://github.com/open-power/HTX/blob/master/Documentation/HTX_user_manual.txt#L36

Reply:
sure